Framework/API/src/AlgorithmHistory.cpp

```diff
@@ -174,6 +174,23 @@ AlgorithmHistory_sptr AlgorithmHistory::operator[](const size_t index) const {
 }
 
 /**
+* Gets the value of a specified algorithm property
+* @param name ::  The property to find
+* @returns The string value of the property
+* @throw Exception::NotFoundError if the named property is unknown
+*/
+const std::string &
+AlgorithmHistory::getPropertyValue(const std::string &name) const {
+  for (auto &hist : m_properties) {
+    if (hist->name() == name) {
+      return hist->value();
+    }
+  }
```

Comment:
Ah, I hadn't considered embedding an anonymous fn.

However, personally I think the original is easier to read (you may feel differently), and I would be suprised if there were any performance differences.